pkg/virt-api/rest/subresource.go

```diff
@@ -445,6 +461,31 @@ func (app *SubresourceAPIApp) RestartVMRequestHandler(request *restful.Request,
 		return
 	}
 
+	// Only force restart is supported for now
+	// Check if bodyMap["graceperiod"] is not empty because gracePeriodint64 can be 0 if that is empty
+	if bodyStruct.GracePeriodSeconds != nil {
+		if *bodyStruct.GracePeriodSeconds == 0 {
+			vmiPodname, err := app.findPod(namespace, vmi)
+			if err != nil {
+				response.WriteError(http.StatusForbidden, err)
```

Comment:
You will have to check for `errors.isNotFound`. If it is a `errors.IsNotFound` error, you will have to return `StatusConflict`. Otherwise `StatusInternalServerError`.

Also I would checko n the VMI explicitliy if it is running before even looking for a pod.

Reply:
About checking explicitly if VMI is running or not, it's already checked here 

https://github.com/kubevirt/kubevirt/pull/2703/files#diff-b2a2ac840fa7df47e840a82a10265a3bR454-R461